### PR TITLE
Dispose the trezor connect iframe upon lock

### PR DESCRIPTION
**Pull request by danjm**
Blocked by merge and release of https://github.com/MetaMask/eth-trezor-keyring/pull/113

This PR addresses an error that would occur if MetaMask is locked and then unlocked, when a Trezor account has been added.

While the error was obscured by an issue with Lavamoat, we were able to determine that it was an error being thrown from within TrezorConnect: `TypeError: Cannot add property default, object is not extensible`

This error is thrown when trying to call `TrezorConnect.init` (within the trezor keyring constructor) if the Trezor iframe already exists. We instantiate a new Trezor keyring whenever we unlock, so if a user was to lock and unlock without closing the browser, this issue could happen.

The `TrezorConnect.dispose` method removes the iframe from the dom. That method is exposed on the keyring in the PR linked above.

This PR makes it so that we call that method upon locking MetaMask.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -3518,6 +3518,12 @@ export default class MetamaskController extends EventEmitter {
    * Locks MetaMask
    */
   setLocked() {
+    const [trezorKeyring] = this.keyringController.getKeyringsByType(
+      'Trezor Hardware',
```
Comment (darkwing) [inline app/scripts/metamask-controller.js:3522]: We should use `KEYRING_TYPES.TREZOR` from `shared/constants/hardware-wallets.js` here

Comment (danjm) [reply app/scripts/metamask-controller.js]: good call, added in latest commit

**Comment by darkwing** [comment]
Tested this and worked well! Really great find on the `dispose`!

**Review by darkwing — DISMISSED** [review]

**Review by darkwing — APPROVED** [review]

**Review by PeterYinusa — APPROVED** [review]

**Comment by PeterYinusa** [comment]
LGTM!

**Pull request merged**